spooky-oysters/bangazon
# Km 09 training

## Link to Ticket
---
[Allow developers to access the Training Program resource](https://github.com/spooky-oysters/bangazon/issues/9)

## Description of Proposed Changes
----
- Models for TrainingProgram, Employee, TrainingEmployee, Department

## Steps to Test
----

```sh
git fetch
git checkout km-09-training
dotnet restore
```
Create an environment variable called: BANGAZON_API_DB which links to the file api.db in the project

At that point, create the database file
```sh
dotnet ef database update
dotnet run
```

Go into Postman or REST Client:
1. path is: localhost:5000/api/TrainingProgram/
1. GET using above path - should be blank until your first post request
1. POST by creating a new JSON object
```sh
{
    "name": "Training",
    "startDate": "2018-02-01T00:00:00",
    "endDate": "2018-02-01T00:00:00",
    "maxAttendance": 2000
}
```

1. GET by ID using localhost:5000/api/TrainingProgram/{id}
1. PUT by updating existing  JSON object and using path localhost:5000/api/TrainingProgram/{id}
1. DELETE using localhost:5000/api/TrainingProgram/{id}
1. DELETE using localhost:5000/api/TrainingProgram/{id} but for a record with a start date in the past


## Impacted Areas in Application
----
List general components of the application that this PR will affect:

* Models, Data, DB, Startup, .csproj

## Mentions @username
----
@jasonfigueroa , @Chewieez , @DreRandaci
